data_steward/cdr_cleaner/cleaning_rules/generate_research_device_ids.py

```diff
@@ -0,0 +1,137 @@
+"""
+Generate research_device_ids for each fitbit person_id/device_id pair.
+
+Original Issue: DC-3229
+
+Device_id is required by privacy to be deidentified in RT and CT. This CR updates the mapping table with newly generated
+ research_device_ids.
+"""
+
+# Python Imports
+import logging
+
+# Project Imports
+from common import JINJA_ENV, PIPELINE_TABLES
+from constants.cdr_cleaner.clean_cdr import QUERY, FITBIT
+from cdr_cleaner.cleaning_rules.base_cleaning_rule import BaseCleaningRule
+
+LOGGER = logging.getLogger(__name__)
+
+ISSUE_NUMBERS = ['DC3229']
+
+# Capture new person_id/device_id pairs, generate research_device_ids and update the masking table
+APPEND_MASKING_TABLE_QUERY = JINJA_ENV.from_string("""
+INSERT INTO `{{project_id}}.{{pipeline_tables}}.wearables_device_id_masking` 
+(person_id, device_id, research_device_id, wearable_type, import_date)
+SELECT 
+  d.person_id, 
+  d.device_id,
+  GENERATE_UUID() as research_device_id, 
+  'fitbit' as wearable_type, 
+  CURRENT_DATE() as import_date
+FROM (SELECT DISTINCT person_id, device_id 
+      FROM `{{project_id}}.{{fitbit_dataset}}.device`) d
```

Comment:
It _should_ not be possible for these fields to be NULL but I agree we should consider the case.

I'll not create a mapping for the case that either field is NULL so that the mapping table remains clean. 
Records without a valid person_id should be cleaned in `RemoveNonExistingPids` and records without a 'device_id' should not be given one.

Side note: I believe `RemoveNonExistingPids` is also assuming person_id is not null. I'll ask around, look into this more and might end up adding a check to the qc notebook that runs after the first set of CRs 'fitbit_qc'. A check for all the fitbit tables to have data in their expected fields. I'll first reopen the conversation of requiring the 'required' fields.

Reply:
Reviewing the query I wanted to account for other possible wearable device data. The possibility that the same person_id/device_id pair exists for multiple 'wearable_types'.